tests/ui/parser/issues/issue-111416.stderr

```diff
@@ -1,18 +1,33 @@
-error: invalid `struct` delimiters or `fn` call arguments
-  --> $DIR/issue-111416.rs:2:14
+error: expected identifier, found `:`
+  --> $DIR/issue-111416.rs:2:30
    |
 LL |     let my = monad_bind(mx, T: Try);
-   |              ^^^^^^^^^^^^^^^^^^^^^^
-   |
-help: if `monad_bind` is a struct, use braces as delimiters
+   |                              ^ expected identifier
    |
-LL |     let my = monad_bind { mx, T: Try };
-   |                         ~            ~
```

Comment:
Still sad of losing this suggestion.

Reply:
Would it be possible for me to add on this suggestion along with the named arguments error? something like -
```
if this is a parameter, remove the parameter name
...

if this is a struct, use braces as delimiters
...
```

Reply:
Yes, you could likely include it to. Look at the existing one because there's some subtlety here: this suggestion is only potentially valid if all the previous expressions in the argument list are either an ident (like `mx`) or the type ascription (which is no longer represented in the expression syntax tree, the `T: Try`), so we only emit the that suggestion if when encountering the problem, all prior args are just idents. We would also want to remove the older logic as now it won't trigger anymore.

Reply:
I understand, I'll take a look at the existing solution and try to adapt it to this. Can we go over other changes that should be made to get all the tests to pass?